xlators/mgmt/volfile-server/src/glusterd-handshake.c

```diff
@@ -0,0 +1,306 @@
+/*
+   Copyright (c) 2010-2012 Red Hat, Inc. <http://www.redhat.com>
+   This file is part of GlusterFS.
+
+   This file is licensed to you under your choice of the GNU Lesser
+   General Public License, version 3 or any later version (LGPLv3 or
+   later), or the GNU General Public License, version 2 (GPLv2), in all
+   cases as published by the Free Software Foundation.
+*/
+
+#include <glusterfs/xlator.h>
+#include <glusterfs/defaults.h>
+#include <glusterfs/glusterfs.h>
+#include <glusterfs/syscall.h>
+#include <glusterfs/compat-errno.h>
+
+#include "glusterd.h"
+#include "glusterd-messages.h"
+#include "glusterfs3.h"
+#include "protocol-common.h"
+#include "rpcsvc.h"
+#include "rpc-common-xdr.h"
+
+typedef ssize_t (*gfs_serialize_t)(struct iovec outmsg, void *data);
+
+size_t
+build_volfile_path(char *volume_id, char *path, size_t path_len)
+{
+    struct stat stbuf = {
+        0,
+    };
+    int32_t ret = -1;
+    char *volid_ptr = NULL;
+    xlator_t *this = THIS;
+    glusterd_conf_t *priv = NULL;
+
+    priv = this->private;
+    GF_ASSERT(priv);
+    GF_ASSERT(volume_id);
+    GF_ASSERT(path);
+
+    if (volume_id[0] == '/') {
+        /* Normal behavior */
```

Comment:
I'm not sure I understand this comment...

Reply:
ah! this is copied from existing code. this is **common** to use `mount -t glusterfs server:/volname /mount/point` (here key is `/volname` and we just want `volname`).